Works on test system

cfgmr opened the pull request:
This version of the plugin contains various small fixes. It works on our test server together with the SixPay test account. 


Please review the changes and let me know what you think.

maxfischer2781 commented:
Looks great! Just two things:

- Concerning the formatting style, do you have any objection to the new-style formatting per se? It seems more readable for non-programmers.

  I would switch to using ``str.format`` and replace the lazy mapping with a regular dict. It's probably faster and more maintainable anyway.

- The ``_verify_amount`` step is not a condition for registering the transaction in the [official PayPal plugin](https://github.com/indico/indico-plugins/blob/master/payment_paypal/indico_payment_paypal/controllers.py#L78). While the result is the same now, someone may see the final ``return True`` as a bug. I would remove the ``if ...:`` and use either a bare ``return`` in ``_verify_amount`` or invert the consistency check and remove any ``return``.

If you have no objections, feel free to add these changes yourself. Otherwise I will do so after merging.

cfgmr commented:
> Concerning the formatting style, do you have any objection to the new-style formatting per se? It seems more readable for non-programmers.

No, not at all, I just wanted to preserve as much of the original code as possible. Changed to new style format placeholders.

> I would switch to using str.format and replace the lazy mapping with a regular dict. It's probably faster and more maintainable anyway.

I got rid of the FieldFormatMap class and replaced it with a simple getter for a map. Is that what you suggested?

> The _verify_amount step is not a condition for registering the transaction in the official PayPal plugin. While the result is the same now, someone may see the final return True as a bug. I would remove the if ...: and use either a bare return in _verify_amount or invert the consistency check and remove any return.

Changed it back to the way it was handled by the PayPal plugin. I was not aware that the _verfiy_amount will send an email in case it fails. It is probably a good idea to make this step non-blocking since even small differences (i.e. due to currency conversion) would prevent successful registration after the user has already paid.

cfgmr commented:
I added a new added a new placeholder {registration_title} which is replaced with the title of the RegistationForm for the Registration.

This helps us to distinguish abstract submission registration, conference reduced registration, ...

To make it work in the order description, I had to increase the max_length of the from 50 to 80. I checked with the [SaferPay API docs](https://saferpay.github.io/jsonapi/index.html#Payment_v1_PaymentPage_Initialize), the max length of Payment-Description is 1000 chars.

If you also find this placeholder useful, could you please add it to the [Plugin docs](http://indico-sixpay.readthedocs.io/en/latest/source/configuration.html)?
